script.js

```diff
@@ -1,4 +1,130 @@
+var scissors = "scissors";
+var paper = "paper";
+var stone = "stone";
+var username = "";
+var playerwins = 0
+var computerwins = 0
+var draws = 0
+var normalsps = 'sps'
+var reversesps = 'reversesps'
+
+// main function for scissors paper stone game.
+
+
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  var randomSps = getSpsRandom();
+  console.log(`computer choice is ${randomSps}`);
+  console.log(`player choice is ${input}`);
+  var myOutputValue = ""
+  if (!username) {
+    if (!input) {
+      return `please input a username`
+    }
+
+    username = input;
+    return `thank you ${username}, please input scissors, paper or stone`;
+  }
+
+
+    if (
+    (input == scissors && randomSps == paper) ||
+    (input == paper && randomSps == stone) ||
+    (input == stone && randomSps == scissors)
+) {
+  playerwins = playerwins + 1;
+  console.log(`playerwins ${playerwins}`);
+}
+
+else if  (
+    (input == scissors && randomSps == stone) ||
+    (input == stone && randomSps == paper) ||
+    (input == paper && randomSps == scissors)
+  ) {
+    computerwins = computerwins + 1;
+    console.log(`computerwins ${computerwins}`);
+  }
+
+else if (
+  input == randomSps) {
+    draws = draws + 1;
+    console.log(`draws ${draws}`);
+
+  }
+  
+
+  // if the outcome is a draw.
+
+  if (input == randomSps) {
+    console.log("it's a draw!");
+    myOutputValue = `it's a draw!`;
+  }
+
+  // if outcome is player wins.
+  else if (
+    (input == scissors && randomSps == paper) ||
+    (input == paper && randomSps == stone) ||
+    (input == stone && randomSps == scissors)
+  ) {
+    console.log("you win!");
+    myOutputValue = `you win!`;
+  }
+
+  // if outcome is player loses.
+  else if (
+    (input == scissors && randomSps == stone) ||
+    (input == stone && randomSps == paper) ||
+    (input == paper && randomSps == scissors)
+  ) {
+    console.log("you lost!");
+    myOutputValue = `you lost!`;
+  }
+
+  // if input is an error
+  
+
+  else {
+    console.log("input error, kindly try again, thanks!");
+    console.log(input);
+    console.log(randomSps);
+    return `input error, kindly input a valid choice and try again! thanks!`;
+  }
```

Comment:
nice way of checking input error. Maybe could check the input first before comparing the player SPS and computer SPS by having if ( input != scissors &&  input != stone&&  input != paper)